engine/src/main/java/org/terasology/world/block/entity/BlockEntitySystem.java

```diff
@@ -131,7 +131,14 @@ public void commonDefaultDropsHandling(CreateBlockDropsEvent event, EntityRef en
             entity.send(new OnBlockToItem(item));
 
             if (shouldDropToWorld(event, block, blockDamageModifierComponent, item)) {
-                processDropping(item, location, blockDamageModifierComponent.impulsePower);
+                float impulsePower;
+                if (blockDamageModifierComponent != null) {
+                    impulsePower = blockDamageModifierComponent.impulsePower;
+                } else {
+                    impulsePower = 0;
+                }
```

Comment:
I feel like this would be more readable if the else was removed and `impulsePower` was initialised to `0` at the same time it is declared.

Reply:
Great catch. Must have missed it at the time :)